frontend/src/pages/CreatePipeline/index.tsx

```diff
@@ -0,0 +1,686 @@
+        const entryNode = nodesRef.current.find((node) => node.type === "entryNodeEdit");
+        const exitNode = nodesRef.current.find((node) => node.type === "exitNodeEdit");
+        const lastEdge = edgesRef.current.find((edge) => edge.target === exitNode?.id);
+
+        const data_in_fields = entryNode?.data.dataIn;
+        const data_out_fields = exitNode?.data.dataOut;
+        const queue: string[] = [lastEdge!.source]
+        let index = 0
+        while (index < queue.length) {
+            const node = nodesRef.current.find((nd) => nd.id === queue[index])
+            node?.data.needs.forEach((nodeId: string) => {
+                queue.push(nodeId)
+            })
+            index++;
+        }
+        const visited: string [] = [];
+        const nodesInStepOrder = queue.reverse().filter((nodeId) => {
+            if (visited.includes(nodeId)) return false;
+            visited.push(nodeId)
+            return true;
+        })
+
+        const steps: any[] = []
+        const tags: any[] = []
+        nodesInStepOrder.forEach((nodeId) => {
+            const incomingEdges = edgesRef.current.filter((edge) => edge.target === nodeId)
+            let condition: string = "";
+            incomingEdges.forEach((edge, index) => {
+                if (edge.data.condition !== "")
+                    condition = index === 0 ? condition.concat(`${edge.data.condition}`) : condition.concat(` and  ${edge.data.condition}`)
+            })
+            const node = nodesRef.current.find((nd) => nd.id === nodeId)
+            node?.data.tags.forEach((tag: any) => {
+                if (tags.filter((existingTag) => existingTag.acronym === tag.acronym).length === 0) {
+                    tags.push(tag);
+                }
+            })
+            condition === "" ?
+                steps.push({
+                    identifier: node?.data.identifier,
+                    needs: node?.data.needs,
+                    inputs: node?.data.selectedDataIn,
+                    service_slug: node?.data.service_slug
+                })
+                : steps.push({
+                    identifier: node?.data.identifier,
+                    needs: node?.data.needs,
+                    condition: condition,
+                    inputs: node?.data.selectedDataIn,
+                    service_slug: node?.data.service_slug
+                })
+
+        })
+
+        return JSON.stringify({
+            name: pipeName,
+            slug: pipeSlug,
+            summary: pipeSummary,
+            description: pipeDescription,
+            data_in_fields: data_in_fields,
+            data_out_fields: data_out_fields,
+            tags: tags,
+            steps: steps
+        })
+    }
+
+
+    React.useEffect(() => {
+        const initialNodes: Node<any>[] = []
+        initialNodes.push(createEntryNode())
+        initialNodes.push(createExitNode())
+        setNodes(() => initialNodes);
+    }, []);
```

Comment:
Either add dependencies to the array or remove it, but empty arrays should be avoided